### ISSUE TITLE
Customizable loss history for `approximator`

### ISSUE THREAD

**Issue by jerrymhuang** [stefanradev93/BayesFlow#217]
Currently, the approximator only stores one set of losses for each training epoch. One has to implement and explicitly call a custom callback (based on Keras' base `Callback` class) in order to visualize the finer-grained loss trajectory across more training steps. 

Here is an example of such workaround implementation where the loss function is recorded across all training steps:

```python
class DetailedLossTrajectory(keras.callbacks.Callback):
    def __init__(self):
        super().__init__()
        self.training_losses = []
        self.validation_losses = []

    def on_train_batch_end(self, batch, logs=None):
        # 'logs' is a dictionary containing loss and other metrics
        training_loss = logs.get('loss')
        self.training_losses.append(training_loss)
        
    def on_test_batch_end(self, batch, logs=None):
        validation_loss = logs.get('loss')
        self.validation_losses.append(validation_loss)
```

It needs to be instantiated and called upon training:

```python
# ...

detailed_loss = DetailedLossTrajectory()

history = approximator.fit(
    epochs=10,
    dataset=training_dataset,
    validation_data=validation_dataset,
    callbacks=[detailed_loss]
)
```

Ideally, we should provide an interface that allows the user to access a more detailed loss trajectory by default, and provide them the ability to control the level of detail (i.e., recording loss every n-th training steps).

**Comment by LarsKue** [comment]
Closing this as no longer relevant.
